### PR TITLE
Include the Sleep sort

### PR DESCRIPTION
**Pull request by MatheusMuriel**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sorts/sleep_sort.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+or
+python3 -m doctest -v sleep_sort.py
+For manual testing run:
+python sleep_sort.py
+"""
+
+from time import sleep
+from threading import Timer
+
+def sleep_sort(collection):
```
[the remark this answers is not in the document]
Comment (MatheusMuriel) [reply sorts/sleep_sort.py:22]: Fixed

File: sorts/sleep_sort.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    >>> sleep_sort([]) == sorted([])
+    True
+    >>> sleep_sort([2, 15, 5]) == sorted([2, 15, 5])
+    True
+    >>> sleep_sort([23, 0, 6, 4, 34]) == sorted([23, 0, 6, 4, 34])
+    True
+    """
+    result = []
+    
+    def append_result(x):
```
[the remark this answers is not in the document]
Comment (MatheusMuriel) [reply sorts/sleep_sort.py:43]: Fixed

File: sorts/sleep_sort.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+python3 -m doctest -v sleep_sort.py
+For manual testing run:
+python sleep_sort.py
+"""
+
+from threading import Timer
+from time import sleep
+
+
+def sleep_sort(collection):
```
[the remark this answers is not in the document]
Comment (MatheusMuriel) [reply sorts/sleep_sort.py:22]: Fixed

File: sorts/sleep_sort.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    >>> sleep_sort([]) == sorted([])
+    True
+    >>> sleep_sort([2, 15, 5]) == sorted([2, 15, 5])
+    True
+    >>> sleep_sort([23, 0, 6, 4, 34]) == sorted([23, 0, 6, 4, 34])
+    True
+    """
+    result = []
+
+    def append_result(x):
```
[the remark this answers is not in the document]
Comment (MatheusMuriel) [reply sorts/sleep_sort.py:43]: Fixed

**Comment by cclauss** [comment]
#2352

**Pull request closed without merging**